### PR TITLE
Build client file structure

### PR DESCRIPTION
**Pull request by MohammedSallout**
- Create react app by typescript and make components, routes, pages ..... directory.
- [npm install] to install the dependencies.
- [npm run dev] to run the app.
- [npm run build] for the production.
Relates #7

### CODE REVIEW

**Review by AbdallahAbujazar — APPROVED** [review]

**Review by haitham-akram — APPROVED** [review]

File: client/src/App.tsx
Comment (haitham-akram) [inline client/src/App.tsx:1]: you could add react-router and check if it is working or not... ,

File: client/vite.config.ts
Comment (haitham-akram) [inline client/vite.config.ts:1]: use this proxy so we can send and receive requests from the src 
``export default defineConfig({
  server: {
    proxy: {
      '/api/v1': 'http://localhost:4000/',
    },
  },
  plugins: [react()],
});``

File: client/src/App.tsx
Comment (MohammedSallout) [reply client/src/App.tsx:1]: i installed react-router-dom and checked it, it's work.

File: client/vite.config.ts
Comment (MohammedSallout) [reply client/vite.config.ts:1]: done

File: client/vite.config.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { defineConfig } from 'vite'
+import react from '@vitejs/plugin-react'
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [react()],
+  server: {
+    proxy: {
+      '/api/v1': {
+        target: 'http://localhost:4000',
```
Comment (Mu7ammadAbed) [inline client/vite.config.ts:10]:
```suggestion
        target: 'http://localhost:3000',
```

Comment (MohammedSallout) [reply client/vite.config.ts]: done

File: client/index.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Vite + React + TS</title>
```
Comment (Mu7ammadAbed) [inline client/index.html:7]:
```suggestion
    <title>Petopia</title>
```

File: client/public/vite.svg [hunk 3]
```diff
@@ -0,0 +1 @@
...x2="57.636%" y1="7.652%" y2="78.411%"><stop offset="0%" stop-color="#41D1FF"></stop><stop offset="100%" stop-color="#BD34FE"></stop></linearGradient><linearGradient id="IconifyId1813088fe1fbc01fb467" x1="43.376%" x2="50.316%" y1="2.242%" y2="89.03%"><stop offset="0%" stop-color="#FFEA83"></stop><stop offset="8.333%" stop-color="#FFDD35"></stop><stop offset="100%" stop-color="#FFA800"></stop></linearGradient></defs><path fill="url(#IconifyId1813088fe1fbc01fb466)" d="M255.153 37.938L134.897 252.976c-2.483 4.44-8.862 4.466-11.382.048L.875 37.958c-2.746-4.814 1.371-10.646 6.827-9.67l120.385 21.517a6.537 6.537 0 0 0 2.322-.004l117.867-21.483c5.438-.991 9.574 4.796 6.877 9.62Z"></path><path fill="url(#IconifyId1813088fe1fbc01fb467)" d="M185.432.063L96.44 17.501a3.268 3.268 0 0 0-2.634 3.014l-5.474 92.456a3.268 3.268 0 0 0 3.997 3.378l24.777-5.718c2.318-.535 4.413 1.507 3.936 3.838l-7.361 36.047c-.495 2.426 1.782 4.5 4.151 3.78l15.304-4.649c2.372-.72 4.652 1.36 4.15 3.788l-11.698 56.621c-.732 3.542 3.979 5.473 5.943 2.437l1.313-2.028l72.516-144.72c1.215-2.423-.88-5.186-3.54-4.672l-25.505 4.922c-2.396.462-4.435-1.77-3.759-4.114l16.646-57.705c.677-2.35-1.37-4.583-3.769-4.113Z"></path></svg>
```
Comment (Mu7ammadAbed) [inline client/public/vite.svg:1]:
```suggestion
<svg width="1000" height="894.737" viewBox="0 0 1000 894.737" fill="none" xmlns="http://www.w3.org/2000/svg">
  <path d="M666.702 702.596c20.211 -16.105 40.123 -31.211 59.211 -47.316 61.754 -52 117.596 -109.491 158.737 -180.123 68.772 -118 50.175 -271.807 -43.386 -365.14 -41.421 -41.298 -88.579 -54.316 -143.702 -32.982 -1.842 0.737 -3.772 1.211 -6.105 1.93 -11.807 -23.14 -28.246 -40.088 -57.263 -44.298C664.737 17.333 694.561 6.351 726.789 2.509c68.035 -8.123 122.246 19.526 167.404 68.333 115.561 124.825 129.211 310.474 33.93 458.316 -51.456 79.825 -118.526 144.632 -192.404 202.649a1351.333 1351.333 0 0 1 -225.228 143.211c-5.035 2.544 -13.211 3.684 -17.825 1.298 -132.421 -67.895 -254.246 -150.86 -355.246 -262.316 -47.07 -51.93 -88.895 -108.105 -109.421 -176.702C-14.456 295.404 12.807 168.509 115.474 61.93 178.772 -3.719 255.807 -15.754 338.825 20.825c14.491 6.421 28.351 14.333 42.14 21.404 -6.491 10.439 -12.842 19.158 -17.544 28.702 -4.684 9.579 -7.684 20 -11.263 29.579 -17.281 -8.07 -34.211 -17.193 -51.982 -24.088 -50 -19.386 -95.825 -10.439 -133.877 27.14 -82.667 81.649 -108 180.702 -82.421 293.93 0.807 3.561 2.07 7.281 4.053 10.298 49.018 74.526 106.316 141.175 180.842 190.702 33.228 22.105 69.702 36.386 109.614 40.561 4.86 0.509 9.807 0.702 14.684 0.474 4.035 -0.175 8.018 -1.14 12.509 -3.877 -3.491 -1.158 -6.965 -2.298 -10.439 -3.474 -52.158 -17.614 -98.965 -44.088 -134.614 -87.368 -7.088 -8.596 -11.333 -21.14 -13 -32.456 -2.684 -18.351 12.596 -29.509 28.298 -20.07 68.018 40.877 144.895 52.719 219.754 71.421 20.596 5.14 41.772 9.649 61.053 18.228 47.649 21.228 80.912 58.053 103.105 105.456 2.035 4.439 4.053 8.947 6.965 15.211Z" fill="#8187DC"/>
  <path d="M363.193 495.965c27.965 -46.228 58.772 -87.088 102.561 -116.316 31.649 -21.123 63.123 -21.158 93.491 2 56.088 42.789 92.807 99.947 115.018 166.544 6.281 18.86 8.228 39.509 9.579 59.526 0.596 8.912 -5.263 18.228 -9.211 30.596 -7.053 -9.07 -11.351 -14.895 -15.93 -20.474 -33.018 -40.281 -74.386 -66.14 -125.158 -77.719 -52.193 -11.93 -103.93 -25.965 -155.825 -39.193 -4.281 -1.053 -8.386 -2.842 -14.526 -4.947Z" fill="#8187DC"/>
  <path d="M539.579 204.439c-0.14 -43.193 11.684 -82.772 37.421 -117.193a112.333 112.333 0 0 1 30.737 -27.772c22.614 -13.842 45.912 -7.368 61.088 14.596 11.702 16.877 16.263 36.246 17.333 56.246 2.579 48.737 -8.421 93.947 -38.14 133.053 -7.544 9.947 -17.544 18.772 -28.07 25.439 -23.316 14.737 -47.491 8.579 -63.105 -14.316 -11.877 -17.368 -17.158 -38.947 -17.263 -70.053Z" fill="#8187DC"/>
  <path d="M364.965 155.965c-0.175 -34.035 3.737 -57.579 14.474 -79.526 20.018 -40.877 57.737 -46.316 87.789 -12.404 22.351 25.228 32.368 56.228 37.281 89.018 4.789 31.947 3.965 63.702 -8.825 93.965 -8.421 19.842 -19.333 38.474 -42.877 41.86 -25.07 3.614 -41.667 -11.351 -55.193 -30.175 -24.035 -33.439 -31.965 -71.895 -32.649 -102.737Z" fill="#8187DC"/>
  <path d="M796.211 348.246c-2.737 40.789 -16.123 76.088 -42.105 106.316 -6.07 7.105 -13.298 13.649 -21.088 18.737 -34.737 22.649 -70.702 10.035 -82.491 -30.053 -13.333 -45.333 -4.211 -88.333 21.404 -126.754 10.228 -15.298 24.421 -29.579 39.912 -39.175 29.175 -18.123 59.175 -7.456 72.807 24.474 6.211 14.456 7.842 30.912 11.561 46.439Z" fill="#8187DC"/>
  <path d="M387.281 364.14c-5.544 17.789 -8.895 36.912 -17.281 53.386 -13.474 26.526 -42.123 33.912 -67.123 18.158 -47.439 -29.895 -67.123 -75.561 -65.825 -130.579 0.404 -16.877 4.088 -35 11.228 -50.123 13.404 -28.351 40.649 -35.702 68.509 -21.368 27.491 14.14 43.912 38.561 54.298 66.491 7.404 19.93 10.649 41.404 16.193 64.035Z" fill="#8187DC"/>
</svg>

```

**Review by Mu7ammadAbed — APPROVED** [review]

**Pull request merged**